entrypoint.sh

```diff
@@ -43,7 +43,7 @@ INSTANCE_TYPE_URL=${INSTANCE_TYPE_URL:-http://169.254.169.254/latest/meta-data/i
 INSTANCE_TYPE=$(curl -s "${INSTANCE_TYPE_URL}")
 
 if [ "${DETACH_ASG}" != "false" ]; then
-  ASG_NAME=$(aws --output text --region "${REGION}" autoscaling describe-auto-scaling-instances --instance-ids "${INSTANCE_ID}" |awk '{print $2}')
+  ASG_NAME=$(aws --output text --region "${REGION}" autoscaling describe-auto-scaling-instances --instance-ids "${INSTANCE_ID}" | awk '$1 == "AUTOSCALINGINSTANCES" {print $2}')
```

Comment:
What happens if the first field isn't `AUTOSCALINGINSTANCES`? Then `ASG_NAME` variable will be empty?

Reply:
as it is a text representation of the API and the API is unlikely to change:
https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_DescribeAutoScalingInstances.html

AUTOSCALINGINSTANCES

is just the "object" which is being returned. Now with this `text` representation the LaunchTemplate inside of this object: https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_AutoScalingInstanceDetails.html
is kind of added in a newline.. thats simply the format they have chosen. But I am going to check something..

Reply:
see my second commit

Reply:
> see my second commit

Cool, makes more sense now!